src/XrdTpc/XrdTpcPMarkManager.cc

```diff
@@ -45,37 +45,34 @@ void PMarkManager::startTransfer(XrdHttpExtReq * req) {
 }
 
 void PMarkManager::beginPMarks() {
-  if(!mSocketInfos.empty() && mPmarkHandles.empty()) {
-    // Create the first pmark handle that will be used as a basis for the other handles
-    // if that handle cannot be created (mPmark->Begin() would return nullptr), then the packet marking will not work
-    // This base pmark handle will be placed at the beginning of the vector of pmark handles
+  if(mSocketInfos.empty()) {
+    return;
+  }
+
+  if(mPmarkHandles.empty()) {
+    // Create the first pmark handle
     std::stringstream ss;
     ss << "scitag.flow=" << mReq->mSciTag;
     SocketInfo & sockInfo = mSocketInfos.front();
-    mInitialFD = sockInfo.client.addrInfo->SockFD();
-    std::unique_ptr<XrdNetPMark::Handle> initialPmark(mPmark->Begin(sockInfo.client, mReq->resource.c_str(), ss.str().c_str(), "http-tpc"));
-    if(initialPmark) {
-      // It may happen that the socket attached to the file descriptor is not connected yet. If this is the case the initial
-      // Pmark will be nullptr...
-      mPmarkHandles.emplace(mInitialFD,std::move(initialPmark));
+    auto pmark = mPmark->Begin(sockInfo.client, mReq->resource.c_str(), ss.str().c_str(), "http-tpc");
+    if(pmark) {
+      mPmarkHandles.emplace(sockInfo.client.addrInfo->SockFD(),std::unique_ptr<XrdNetPMark::Handle>(pmark));
       mSocketInfos.pop();
     }
```

Comment:
You are missing a `return` here in case `pmark` creation fails for whatever reason, no?